### PR TITLE
add serverside validation

### PR DESCRIPTION
**Pull request by sarahgibs**
Adds server-side validation for the provided service name when creating a new service with SDARQ. Also adds a check to make sure that the product ID returned by DefectDojo is an int.

### CODE REVIEW

File: sdarq/backend/src/dojo_helper.py [hunk 1]
```diff
@@ -22,10 +22,14 @@ def dojo_create_or_update(name, description, product_type, user_email, appsec_sl
     response = requests.get(products_endpoint+"?name_exact="+name, headers=headers)
     if response.json()['count'] > 0:
         product_id = response.json()['results'][0]['id']
+        if not isinstance(product_id, int):
+            #this should never happen
```
Comment (TomConner) [inline sdarq/backend/src/dojo_helper.py:26]: probably should log if it does

File: sdarq/backend/src/app.py [hunk 2]
```diff
@@ -168,7 +168,9 @@ def submit():
         del json_data['Ticket_Description']
 
         product_id = dojo_helper.dojo_create_or_update(dojo_name, parse_json_data.prepare_dojo_input(json_data), product_type, user_email, appsec_slack_channel, security_champion, dojo_host_url)
-
+        if not product_id:
+            logging.error('DefectDojo did not respond correctly when attempting to update the product')
```
Comment (TomConner) [inline sdarq/backend/src/app.py:172]: if the raise message will be logged then you don't need logging.error here; if not then you need it at 233.

File: sdarq/backend/src/app.py [hunk 3]
```diff
@@ -226,6 +228,9 @@ def submit_app():
     try:
         json_data = request.get_json()
         dojo_name = json_data['Service']
+        regex = "^[a-zA-Z0-9][a-zA-Z0-9-_ ]{1,40}[a-zA-Z0-9]$"
+        if not (re.search(regex,dojo_name)):
+            raise ValueError("The provided project name was not well formated.")
```
Comment (TomConner) [inline sdarq/backend/src/app.py:233]:
```suggestion
            raise ValueError("The provided project name was not well formatted.")
```

File: sdarq/backend/src/app.py [hunk 4]
```diff
@@ -226,6 +228,9 @@ def submit_app():
     try:
         json_data = request.get_json()
         dojo_name = json_data['Service']
+        regex = "^[a-zA-Z0-9][a-zA-Z0-9-_ ]{1,40}[a-zA-Z0-9]$"
+        if not (re.search(regex,dojo_name)):
```
Comment (TomConner) [inline sdarq/backend/src/app.py:232]: add logging.error here, or remove from line 172?

File: sdarq/backend/src/dojo_helper.py [hunk 5]
```diff
@@ -22,10 +22,14 @@ def dojo_create_or_update(name, description, product_type, user_email, appsec_sl
...
+            return None
         data['description'] = data['description'] + " updated by " + user_email
         res = requests.patch(products_endpoint + str(product_id) + "/", headers=headers, data=json.dumps(data))
         if res.status_code != 200:
             logging.info("failed to update product")
```
Comment (TomConner) [inline sdarq/backend/src/dojo_helper.py:31]:
```suggestion
            logging.error("failed to update product")
```
but you may want to ignore this suggestion as you didn't edit this line

**Review by TomConner — APPROVED** [review]
Validation is good!

(a couple little inline comments to address or not as you see fit)

File: sdarq/backend/src/app.py (hunk 4 above)
Comment (sarahgibs) [reply sdarq/backend/src/app.py:232]: the value error that is on the next line gets logged.

File: sdarq/backend/src/dojo_helper.py (hunk 1 above)
Comment (sarahgibs) [reply sdarq/backend/src/dojo_helper.py:26]: it logs the exception message.

**Review by TomConner — APPROVED** [review]

**Pull request merged**